tensorflow/noscope/diff_detector.h

```diff
@@ -0,0 +1,51 @@
+#include "filter.h"
+
+namespace noscope {
+
+class DiffDetector : public Filter {

```

Comment:
Should probably have a virtual difference detector class and subclass it. Let's discuss offline